app/src/behaviors/behavior_sticky_key.c

```diff
@@ -247,14 +265,33 @@ static int sticky_key_keycode_state_changed_listener(const zmk_event_t *eh) {
                 sticky_key->modified_key_usage_page == ev_copy.usage_page &&
                 sticky_key->modified_key_keycode == ev_copy.keycode) {
                 stop_timer(sticky_key);
-                release_sticky_key_behavior(sticky_key, ev_copy.timestamp);
+                sticky_keys_to_release_after_reraise[i] = sticky_key;
             }
         }
     }
-    if (event_reraised) {
-        return ZMK_EV_EVENT_CAPTURED;
+
+    // give each sticky key a chance to press their behavior before the event is reraised
+    for (int i = 0; i < ZMK_BHV_STICKY_KEY_MAX_HELD; i++) {
+        struct active_sticky_key *sticky_key = sticky_keys_to_press_before_reraise[i];
+        if (sticky_key) {
+            press_sticky_key_behavior(sticky_key, ev_copy.timestamp);
```

Comment:
Do we need to clear if from `sticky_keys_to_press_before_reraise` now that it's pressed?

Reply:
It will only loop once through the array before discarding it (and it is null-initialized). The purpose is simply to collect all the out-of-order presses (of just-in-time mods) and releases (of expired mods) and execute them in order "around" the other key.